commit 3d3d91da452b4d347d314587db301b64b3f90e20
Author: bastimeyer <mail@bastimeyer.de>
Date:   Wed Feb 9 15:32:40 2022 +0100

    effects/desktopgrid: fix paint screen mask
    
    Fix zoom-in/zoom-out animation overlaying on top of other screens

diff --git a/src/effects/desktopgrid/desktopgrid.cpp b/src/effects/desktopgrid/desktopgrid.cpp
index e444d9ff4..ee0d9b18c 100644
--- a/src/effects/desktopgrid/desktopgrid.cpp
+++ b/src/effects/desktopgrid/desktopgrid.cpp
@@ -228,8 +228,10 @@ void DesktopGridEffect::prePaintScreen(ScreenPrePaintData& data, std::chrono::mi
         }
         // PAINT_SCREEN_BACKGROUND_FIRST is needed because screen will be actually painted more than once,
         // so with normal screen painting second screen paint would erase parts of the first paint
-        if (timeline.currentValue() != 0 || (isUsingPresentWindows() && isMotionManagerMovingWindows()))
-            data.mask |= PAINT_SCREEN_TRANSFORMED | PAINT_SCREEN_BACKGROUND_FIRST;
+        if (timeline.currentValue() != 0 || (isUsingPresentWindows() && isMotionManagerMovingWindows())) {
+            int paint = activated && timeline.currentValue() == 1 ? PAINT_SCREEN_TRANSFORMED : PAINT_SCREEN_REGION;
+            data.mask |= paint | PAINT_SCREEN_BACKGROUND_FIRST;
+        }
     }
 
     const EffectWindowList windows = effects->stackingOrder();
